primitive/crypto/src/arithmetic/bits_256/coordinate/projective.rs

```diff
@@ -11,6 +12,12 @@ pub fn add_point(
     inv: u64,
 ) -> ProjectiveCoordinate<[u64; 4]> {
     let zero: [u64; 4] = [0; 4];
+    if lhs == (zero, zero, zero) {
+        return rhs;
+    } else if rhs == (zero, zero, zero) {
+        return lhs;
+    }
+
```

Comment:
Do we have the case that z coordinate is zero but the others are not zero?
If we have, it's going to be a problem.